qt/python/instrumentview/instrumentview/FullInstrumentViewPresenter.py

```
@@ -91,21 +91,25 @@ def __init__(self, view: FullInstrumentViewWindow, model: FullInstrumentViewMode
-            func, args = item
+            should_stop = False
             try:
-                if not self._closing:
-                    func(*args)
+                if item is self._callback_stop_sentinel:
+                    should_stop = True
+                else:
+                    func, args = item
+                    if not self._closing:
+                        func(*args)
             except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)